### PR TITLE
data_module: Add guards that ensures only valid configs are applied

### PR DESCRIPTION
**Pull request by simensrostad**

### CODE REVIEW

File: src/modules/data_module.c [hunk 1]
```diff
@@ -621,43 +623,56 @@ static void on_cloud_state_connected(struct data_msg_data *msg)
...
+		    new.actw > current_cfg.gpst + 10 &&
+		    new.actw > 0) {
 			current_cfg.actw = new.actw;
 			LOG_WRN("New Active timeout: %d", current_cfg.actw);
 
 			config_change = true;
 		}
 
-		if (current_cfg.pasw != new.pasw && new.pasw != 0) {
+		if (current_cfg.pasw != new.pasw &&
```
Comment (jtguggedal) [inline src/modules/data_module.c]: I vote for some parentheses in these ifs to make if obvious at first glance what's being compared

File: src/modules/data_module.c [hunk 2]
```diff
@@ -621,43 +623,56 @@ static void on_cloud_state_connected(struct data_msg_data *msg)
...
+		}
+
+		/* Only apply a new active wait timeout configuration if the
+		 * new active wait timeout exceeds the current GPS timeout value
+		 * + 10 seconds. This is to hinder requesting GPS data from the
+		 * GPS module in the middle of a GPS search.
+		 */
+		if (current_cfg.actw != new.actw &&
+		    new.actw > current_cfg.gpst + 10 &&
+		    new.actw > 0) {
```
Comment (jtguggedal) [inline src/modules/data_module.c]: Perhaps print warnings if incoming configs are out of bounds?

File: src/modules/data_module.c (hunk 1 above)
Comment (simensrostad) [reply src/modules/data_module.c]: Yes. Lets do that.

File: src/modules/data_module.c (hunk 2 above)
Comment (simensrostad) [reply src/modules/data_module.c]: Agreed.

**Review by jtguggedal — APPROVED** [review]

**Pull request merged**